dht/range_streamer.cc

```diff
@@ -224,10 +224,12 @@ bool range_streamer::use_strict_sources_for_ranges(const sstring& keyspace_name)
     auto& strat = ks.get_replication_strategy();
     auto rf = strat.get_replication_factor();
     auto nr_nodes_in_ring = get_token_metadata().get_all_endpoints().size();
+    bool everywhere_topology = strat.get_type() == locator::replication_strategy_type::everywhere_topology;
     // Use strict when number of nodes in the ring is equal or more than RF
     auto strict = !_db.local().is_replacing()
            && _db.local().get_config().consistent_rangemovement()
            && !_tokens.empty()
+           && !everywhere_topology
```

Comment:
I'm worried if we won't lose data this way.
What will happen if we wrote with cl=one and then decommission the only node that got the data? Will data be streamed from this node to some other one?

Or another example: suppose we have 3 nodes and we wrote with quorum, the data ended up on 2 nodes. Then we add 4th node, which node will it stream data from? Suppose it streams from the node which didn't get the data. Then we have 2 nodes with and 2 nodes without data. Now bootstrap 5th node, if it's possible to stream from the node which didn't get the data then we have 3 nodes without data and 2 nodes with data. Now if we read with quorum we can miss the data.

Reply:
This is the old stream based bootstrap which streams from only one node.

Reply:
For non everywhere strategy, we will stream from the node that is losing the range, so it is safe. For everywhere strategy, no node will lose range, we will stream from any node.

Reply:
that's the reason repair-based node operations were added.